runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9609,6 +9609,42 @@ J9::Z::TreeEvaluator::VMinlineCallEvaluator(TR::Node * node, bool indirect, TR::
    return callWasInlined;
    }
 
+void
+J9::Z::TreeEvaluator::genGuardedLoadOOL(TR::Node *node, TR::CodeGenerator *cg, TR::Register *byteSrcReg, TR::Register *byteDstReg, TR::Register *byteLenReg, TR::LabelSymbol *mergeLabel, TR_S390ScratchRegisterManager *srm, bool isForward)
+   {
+   TR_ASSERT_FATAL(J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE == 0x20000,
+               "GSCS: The OOL sequence branch is dependant on the flag being 0x20000");
+   TR_ASSERT_FATAL(TR::Compiler->target.is64Bit(),
+               "GSCS: Guarded Load OOL Path only defined in 64 bit mode");
+
+   TR::LabelSymbol* slowPathLabel = generateLabelSymbol(cg);
+
+   TR::Register *vmReg = cg->getMethodMetaDataRealRegister();
+   TR::MemoryReference *privFlagMR = generateS390MemoryReference(vmReg,
+   TR::Compiler->vm.thisThreadGetConcurrentScavengeActiveByteAddressOffset(cg->comp()), cg);
+
+   generateSIInstruction(cg, TR::InstOpCode::TM, node, privFlagMR, 0x00000002);
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_CC3, node, slowPathLabel);
+   // Generate OOL Slow Path
+   TR_S390OutOfLineCodeSection* outOfLineCodeSection = new (cg->trHeapMemory()) TR_S390OutOfLineCodeSection(slowPathLabel, mergeLabel, cg);
```

Comment:
Please avoid using placement `new` as there are APIs to handle this. Similarly below for dependencies.

Reply:
@fjeremic I looked at the available API for OutOfLine Code Section. I could not find one that generates `TR_S390OutOfLineCodeSection` . Also for the register dependency, I think we need to add more `generate*` to the API . In this case I need to construct new Dependency using available deps. We have constructor to do that but no `generate*` API. We should certainly add new functions to RegisterDependency API but that is out of this PR's scope and we need separate PR which adds new functions and updates code.

Reply:
Ah you're right!